arch/inst/C/c.ldsp.yaml

```diff
@@ -36,6 +45,9 @@ operation(): |
     raise(ExceptionCode::IllegalInstruction, mode(), $encoding);
   }
 
-  XReg virtual_address = X[2] + imm;
+  if (xlen() == 32) {
+    raise(ExceptionCode::IllegalInstruction, mode(), $encoding);
+  }
```

Comment:
Like the others, the 32 bit code needs to be merged in here, and this test deleted/modified such that you support both 64 bit mode (as written below), and 32 bit mode (using register pairs).